server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -161,7 +164,16 @@ public void configure(String configuration) {
   @Override
   public Status replicate(final Path p, final Status status, final ReplicationTarget target, final ReplicaSystemHelper helper) {
     final Instance localInstance = HdfsZooInstance.getInstance();
-    final AccumuloConfiguration localConf = new ServerConfigurationFactory(localInstance).getConfiguration();
+    AccumuloConfiguration instanceConf = new ServerConfigurationFactory(localInstance).getConfiguration();
+
+    // Overwrite RPC timeout value used by ClientContext with the replication timeout value
+    Map<String,String> confMap = new HashMap<>();
+    instanceConf.getProperties(confMap, Predicates.<String> alwaysTrue());
+    confMap.put(Property.GENERAL_RPC_TIMEOUT.getKey(), instanceConf.get(Property.REPLICATION_RPC_TIMEOUT));
```

Comment:
I'd be in favor of that if you'd like to roll that change in too. Looks like there's precedence in that class already https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java#L177